File: py/Ram.py
from Komponen import Komponen

# Ram.py
class Ram(Komponen):
    def __init__(self, kapasitasGB=0, ddr="", merk="", nama=""):
        super().__init__(merk, nama)
        self.kapasitasGB = kapasitasGB
        self.ddr = ddr
    
    def setKapasitasGB(self, kapasitasGB):
        self.kapasitasGB = kapasitasGB
    
    def setDdr(self, ddr):
        self.ddr = ddr
    
    def getKapasitasGB(self):
        return self.kapasitasGB
    
    def getDdr(self):
        return self.ddr

File: cpp/Ram.cpp
// Ram.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Ram : public virtual Komponen
{
private:
    int kapasitasGB;
    string ddr;
public:
    Ram() {}

    Ram(int kapasitasGB, string ddr, string merk, string nama)
        : Komponen(merk, nama), kapasitasGB(kapasitasGB), ddr(ddr) {}

    void setKapasitasGB(int kapasitasGB) { this->kapasitasGB = kapasitasGB; }
    void setDdr(string ddr) { this->ddr = ddr; }

    int getKapasitasGB() const { return kapasitasGB; }
    string getDdr() const { return ddr; }

    ~Ram() {}
};
